src/hotspot/share/services/mallocTracker.cpp

```diff
@@ -289,3 +290,32 @@ void* MallocTracker::record_free(void* memblock) {
   header->release();
   return (void*)header;
 }
+
+// Given a pointer, if it seems to point to the start of a valid malloced block,
+// print the block. Note that since there is very low risk of memory looking
+// accidentally like a valid malloc block header (canaries and all) this is not
+// totally failproof. Only use this during debugging or when you can afford
+// signals popping up, e.g. when writing an hs_err file.
+bool MallocTracker::print_pointer_information(const void* p, outputStream* st) {
+  assert(MemTracker::enabled(), "NMT must be enabled");
+  if (CanUseSafeFetchN() && os::is_readable_pointer(p)) {
```

Comment:
`os::is_readable_pointer()` uses `CanUseSafeFetch32()`, you may want to check `CanUseSafeFetch32()` instead of `CanUseSafeFetchN()`.

Reply:
Good point. Done.